Makefile

```diff
@@ -337,13 +343,36 @@ endif
 
 
 ### This target auto-generates a new grub.cfg file and uses grub to build a bootable ISO.
-### This target should be invoked when all of contents of `GRUB_ISOFILES` are ready to be packaged into an ISO.
+### This target should be invoked when all of contents of `ISOFILES` are ready to be packaged into an ISO.
 grub:
-	@cargo run --release --manifest-path $(ROOT_DIR)/tools/grub_cfg_generation/Cargo.toml -- $(GRUB_ISOFILES)/modules/ -o $(GRUB_ISOFILES)/boot/grub/grub.cfg
-	@$(GRUB_MKRESCUE) -o $(iso) $(GRUB_ISOFILES)  2> /dev/null
-
-
-### This target copies all extra files into the `GRUB_ISOFILES` directory,
+	@mkdir -p $(ISOFILES)/boot/grub
+	@cargo run --release --manifest-path $(ROOT_DIR)/tools/grub_cfg_generation/Cargo.toml -- $(ISOFILES)/modules/ -o $(ISOFILES)/boot/grub/grub.cfg
+	@$(GRUB_MKRESCUE) -o $(iso) $(ISOFILES)  2> /dev/null
+
+### This target uses limine to build a bootable ISO.
+### This target should be invoked when all of contents of `ISOFILES` are ready to be packaged into an ISO.
+limine-iso:
+	@test -d limine/ || echo "Missing ./limine/ directory! Did you follow the Readme?"
+	@cd $(OBJECT_FILES_BUILD_DIR)/ && ls | cpio --no-absolute-filenames -o > $(ISOFILES)/modules.cpio
+	@cargo run -r --manifest-path $(ROOT_DIR)/tools/limine_compress_modules/Cargo.toml -- -i $(ISOFILES)/modules.cpio -o $(ISOFILES)/modules.cpio.lz4
+	@rm -r $(OBJECT_FILES_BUILD_DIR)
+	@rm $(ISOFILES)/modules.cpio
+	@cp cfg/limine.cfg $(ISOFILES)/
+	@cp limine/limine-cd.bin $(ISOFILES)/
+	@cp limine/limine-cd-efi.bin $(ISOFILES)/
+	@cp limine/limine.sys $(ISOFILES)/
+	tree $(ISOFILES)/
+	@rm -f $(iso)
+	@xorriso -as mkisofs \
+		-b limine-cd.bin -no-emul-boot -boot-load-size 4 \
+		-boot-info-table --efi-boot limine-cd-efi.bin \
+		-efi-boot-part --efi-boot-image --protective-msdos-label \
+		$(ISOFILES)/ -o $(iso)
+	@$(MAKE) -C limine
```

Comment:
using `LIMINE_DIR` would help clarify this too, I mistakenly thought `limine` was a new target when reading this.

Reply:
done